runtime/oti/j9nonbuilder.h

```diff
@@ -5669,6 +5669,10 @@ typedef struct J9JavaVM {
 	struct J9HashTable* ensureHashedClasses;
 #if JAVA_SPEC_VERSION >= 19
 	U_64 nextTID;
+	j9object_t *liveVirtualThreadList;
```

Comment:
This is the root of the linked list, which will be a `VirtualThread` object ref. Assuming, these objects are not GC roots. If the GC moves this object, then the object ref stored in `liveVirtualThreadList` will become stale. We may need to store the root of the linked list as a `static field` in the `VirtualThread` class. I didn't find any code to add static fields dynamically during runtime.

Reply:
looking through the j9objects in `J9JavaVM` struct, standard practise seem to be using `j9jni_createGlobalRef` to generate a global ref and store that as a `j9object_t*` in the vm, if so, we may need to create a empty vthread as the root entry to avoid have to recreate global ref every time the root entry completes and gets removed

Reply:
You could also update the contents of the ref when the list changes, though there may be some barrier implications for GC.

Reply:
side note: will looking through the vm struct, `destroyVMState` and `selectorHashTable` seem to be legacy code with no reference, maybe those can be cleaned up as well.

Reply:
Can I initialize it in `stdinit.c::standardInit()`? Will there be any crashes from VirtualThread static initializers trying to fetch system properties?

Reply:
```
notifyJvmtiMountBegin {
   if (null == liveVirtualThreadList) {
       create an empty virtual thread using vm->memoryManagerFunctions->J9AllocateObject
       create a j9jni_createGlobalRef
   }

terminateVMThreading {
   remove the empty virtual thread
   remove the corresponding j9jni_createGlobalRef
```

> Can I initialize it in `stdinit.c::standardInit()`?

It will add to the startup cost: loading an extra class + creating an object; especially when virtual threads will never be used. Only minimal changes which are critical to start the VM should go in `stdinit.c::standardInit()`. `VirtualThread` init is not considered critical to VM startup. Even though, we can do the operation in `stdinit.c::standardInit()`; we should avoid it since it adds more overhead to the startup cost.